downstream/modules/platform/proc-controller-create-credential-type.adoc

```diff
@@ -0,0 +1,214 @@
+},{
+    # field 2...
+},{
+    # field 3...
+}],
+
+"required": ["api_token"]            # optional; one or more fields can be marked as required
+},
+----
++
+When `type=string`, fields can optionally specify multiple choice options:
++
+[literal, options="nowrap" subs="+attributes"]
+----
+{
+  "fields": [{
+      "id": "api_token",          # required - a unique name used to reference the field value
+      "label": "API Token",       # required - a unique label for the field
+      "type": "string",
+      "choices": ["A", "B", "C"]
+  }]
+},
+----
+
+. In the *Injector Configuration* field, enter environment variables or extra variables that specify the values a credential type can inject.
+The format can be in YAML or JSON (see examples in the previous step).
++
+The configuration in JSON format below show each field and how they are used:
++
+[literal, options="nowrap" subs="+attributes"]
+----
+{
+  "file": {
+      "template": "[mycloud]\ntoken={{ api_token }}"
+  },
+  "env": {
+      "THIRD_PARTY_CLOUD_API_TOKEN": "{{ api_token }}"
+  },
+  "extra_vars": {
+      "some_extra_var": "{{ username }}:{{ password }}"
+  }
+}
+----
++
+Credential Types can also generate temporary files to support `.ini` files or certificate or key data:
++
+[literal, options="nowrap" subs="+attributes"]
+----
+{
+  "file": {
+      "template": "[mycloud]\ntoken={{ api_token }}"
+  },
+  "env": {
+      "MY_CLOUD_INI_FILE": "{{ tower.filename }}"
+  }
+}
+----
++
+In this example, {ControllerName} writes a temporary file that contains:
++
+[literal, options="nowrap" subs="+attributes"]
+----
+[mycloud]\ntoken=SOME_TOKEN_VALUE
+----
++
+The absolute file path to the generated file is stored in an environment variable named `MY_CLOUD_INI_FILE`.
++
+The following is an example of referencing multiple files in a custom credential template:
++
+*Inputs*
++
+[literal, options="nowrap" subs="+attributes"]
+----
+{
+  "fields": [{
+    "id": "cert",
+    "label": "Certificate",
+    "type": "string"
+  },{
+    "id": "key",
+    "label": "Key",
+    "type": "string"
+  }]
+}
+----
++
+*Injectors*
++
+[literal, options="nowrap" subs="+attributes"]
+----
+{
+  "file": {
+    "template.cert_file": "[mycert]\n{{ cert }}",
+    "template.key_file": "[mykey]\n{{ key }}"
+},
+"env": {
+    "MY_CERT_INI_FILE": "{{ tower.filename.cert_file }}",
+    "MY_KEY_INI_FILE": "{{ tower.filename.key_file }}"
+}
+}
+----
+
+. Click btn:[Save].
+. Your newly created credential type appears on the list of credential types:
++
+image:credential-types-new-listed.png[New credential type]
+
+. Click the image:leftpencil.png[Edit,15,15] icon to modify the credential type options.
++
+[NOTE]
+====
+In the *Edit* screen, you can modify the details or delete the credential.
+If the *Delete* button is disabled, it is indication that the credential type is being used by a credential, and you must delete the credential type from all the credentials that use it before you can delete it. 
```

Comment:
Avoid "button", suggested re-write "If the *Delete* option is disabled.." or " If *Delete* is disabled.."